### ISSUE TITLE
VT as proposal distribution for the calculation of expected rate

### ISSUE THREAD

**Issue by Qazalbash** [gwkokab/gwkokab#381]
VT assigns a number to each point in the parameter space. A normalized VT can be interpreted as a probability distribution over the parameter space. We can use this distribution as the proposal distribution for the importance sampling method. The Expected Rate can be estimated as,

$$
\mu_{\Omega\mid\Lambda}\approx\frac{1}{N}\sum_{1\leq i\leq N,\omega_i\sim\text{VT}}\rho_{\Omega\mid\Lambda}(\omega_i\mid\lambda),
$$

where $N$ is the number of samples drawn from the proposal distribution, we only require samples from normalized VT. This is a one-time process and the samples can be stored for further computations.

**Comment by Qazalbash** [comment]
An example script to generate samples from VT.

```python
import numpy as np
import numpyro
from gwkokab.parameters import PRIMARY_MASS_SOURCE, SECONDARY_MASS_SOURCE
from gwkokab.vts import load_model
from jax import nn as jnn, numpy as jnp, random as jrd, vmap
from numpyro import distributions as dist
from numpyro.infer import MCMC, NUTS


vt_filename = "/media/gradf/academia/kanals/vts/vt_0p5_200_IMRPhenomPv2_SimNoisePSDaLIGO175MpcT1800545.eqx"

_, logVT = load_model(vt_filename)


def model():
    m1 = numpyro.sample(
        PRIMARY_MASS_SOURCE.name, dist.Uniform(mmin, mmax, validate_args=True)
    )
    m2 = numpyro.sample(
        SECONDARY_MASS_SOURCE.name, dist.Uniform(mmin, mmax, validate_args=True)
    )
    m1m2 = jnp.stack([m1, m2], axis=-1)
    numpyro.factor("log_vt_fac", logVT(m1m2))


nuts = NUTS(model, target_accept_prob=0.99)

mcmc = MCMC(
    nuts, num_warmup=10000, num_chains=10, num_samples=1000, jit_model_args=True
)

mcmc.run(jrd.PRNGKey(26))

mcmc.print_summary()

samples = mcmc.get_samples()


np.savetxt(
    "vt_0p5_200_IMRPhenomPv2_SimNoisePSDaLIGO175MpcT1800545.dat",
    jnp.stack(
        [samples[PRIMARY_MASS_SOURCE.name], samples[SECONDARY_MASS_SOURCE.name]],
        axis=-1,
    ),
    header=" ".join([PRIMARY_MASS_SOURCE.name, SECONDARY_MASS_SOURCE.name]),
)
```
The left plot contains samples generated by the above script and the right plot is the heat map of VT (yellow is high probability and violet is low).

![vt_samples](https://github.com/user-attachments/assets/4a8c3914-252b-4b5f-a328-c84f912560ac)
